src/xkernel.cpp

```diff
@@ -74,14 +75,190 @@ namespace xpyt
 
         return kernel_module;
     }
+}
+
+
+namespace xpyt_raw
+{
+
+    struct xmock_ipython
+    {
+        py::object kernel;
+        void register_post_execute(py::args, py::kwargs) {};
+        void enable_gui(py::args, py::kwargs) {};
+        void observe(py::args, py::kwargs) {};
+        void showtraceback(py::args, py::kwargs) {};
+    };
 
-    py::module get_kernel_module()
+    struct xmock_kernel
     {
-        static py::module kernel_module = get_kernel_module_impl();
+        xmock_kernel() = default;
+
+        inline py::object parent_header() const
+        {
+            return py::dict(py::arg("header") = xeus::get_interpreter().parent_header().get<py::object>());
+        }
+
+        xpyt::xcomm_manager m_comm_manager;
+    };
+
+    /*****************
+     * kernel module *
+     *****************/
+
+    void bind_history_manager(py::module& kernel_module)
+    {
+        py::class_<xeus::xhistory_manager>(kernel_module, "HistoryManager")
```

Comment:
I am not sure we actually need this, IIRC this was for the custom magics we had before depending on IPython. But let's keep it for now and see later if we can remove it.